pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -104,6 +104,11 @@ func NewCodec(
 			}
 			internal.encodeVersion[v.Group] = v
 		}
+		if len(internal.encodeVersion) == 1 {
+			for _, v := range internal.encodeVersion {
```

Comment:
We need to get the first value out of the map without knowing the key.

On Mon, May 23, 2016 at 5:31 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/runtime/serializer/versioning/versioning.go
> <https://github.com/kubernetes/kubernetes/pull/26007#discussion_r64192958>
> :
>
> > @@ -104,6 +104,11 @@ func NewCodec(
> >  			}
> >  			internal.encodeVersion[v.Group] = v
> >  		}
> > +		if len(internal.encodeVersion) == 1 {
> > +			for _, v := range internal.encodeVersion {
>
> I don't understand it - why do you want to have a for if there is exactly
> one element in the slice?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26007/files/b1a759cbbc594a7552971a09896034298daf114a..a9c3edc61300320016c2820e28b17260a548118e#r64192958>
>